sql/core/src/test/scala/org/apache/spark/sql/execution/CoalesceShufflePartitionsSuite.scala

```diff
@@ -408,6 +412,25 @@ class CoalesceShufflePartitionsSuite extends SparkFunSuite with BeforeAndAfterAl
     }
     withSparkSession(test, 100, None)
   }
+
+  test("SPARK-34790: enable IO encryption in AQE partition coalescing") {
```

Comment:
I can't reproduce the issue by this test. @hezuojiao Could you double-check it?

Reply:
I can reproduce the issue because my laptop has only 4 cores :) 
I updated the test case by using fixed `numInputPartitions` instead of `sparkContext.defaultParallelism` which is related to the number of computer cores.
You can try to run this test again now. @Ngone51